samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -517,23 +510,25 @@ class OffsetManager(
     if (startpointManager != null) {
       info("Starting startpoint manager.")
       startpointManager.start
-      val taskNameToSSPs: Map[TaskName, Set[SystemStreamPartition]] = systemStreamPartitions
 
-      taskNameToSSPs.foreach {
+      systemStreamPartitions.foreach {
         case (taskName, systemStreamPartitionSet) => {
-          val sspToStartpoint = systemStreamPartitionSet
-            .map(ssp => (ssp, startpointManager.readStartpoint(ssp, taskName)))
-            .filter(_._2 != null)
-            .toMap
-
-          if (!sspToStartpoint.isEmpty) {
-            startpoints += taskName -> sspToStartpoint
+          Option(startpointManager.getFanOutForTask(taskName)) match {
+            case Some(fanOut) => {
+              val filteredFanOut = fanOut.asScala
+                .filter(f => systemStreamPartitionSet.contains(f._1))
+                .toMap
+              if (!filteredFanOut.isEmpty) {
+                startpoints += taskName -> filteredFanOut
```

Comment:
Do you have a log somewhere which shows that there was a fanout entry?

Reply:
Only when creating the fan outs. Will add a log line here.